src/widgets/vitals/vitals-biometrics-form/vitals-biometrics-form.component.tsx

```diff
@@ -0,0 +1,383 @@
+
+  const [
+    bloodPressureUnit,
+    ,
+    temperatureUnit,
+    heightUnit,
+    weightUnit,
+    pulseUnit,
+    oxygenSaturationUnit,
+    midUpperArmCircumferenceUnit,
+    respiratoryRateUnit
+  ] = conceptsUnits;
+
+  const isBMIInValidRange = value => {
+    if (value === undefined || value === "") return false;
+    return !(value >= 18.5 && value <= 24.9);
+  };
+
+  closeWorkspace = closeWorkspace ?? (() => switchTo("workspace", ""));
+
+  const handleCancel = () => {
+    closeWorkspace();
+  };
+
+  const savePatientVitalsAndBiometrics = () => {
+    const ac = new AbortController();
+    savePatientVitals(
+      config.vitals.encounterTypeUuid,
+      config.vitals.formUuid,
+      config.concepts,
+      patientUuid,
+      patientVitalAndBiometrics,
+      new Date(),
+      ac,
+      session.sessionLocation.uuid
+    ).then(response => {
+      response.status === 201 && closeWorkspace();
+      response.status !== 201 && createErrorHandler();
+    });
+    return () => ac.abort();
+  };
+
+  useEffect(() => {
+    if (
+      patientVitalAndBiometrics?.height &&
+      patientVitalAndBiometrics?.weight
+    ) {
+      const calculatedBmi = calculateBMI(
+        Number(patientVitalAndBiometrics.weight),
+        Number(patientVitalAndBiometrics.height)
+      );
+      setPatientBMI(calculatedBmi);
+    }
+  }, [patientVitalAndBiometrics?.weight, patientVitalAndBiometrics?.height]);
+
+  return (
+    <BrowserRouter basename={window["getOpenmrsSpaBase"]()}>
+      <Grid condensed className={styles.vitalsBiometricContainer}>
+        <Row>
+          <Column>
+            <p className={styles.vitalsTitle}>Vitals</p>
+          </Column>
+        </Row>
+        <Row>
+          <Column>
+            <VitalsBiometricInput
+              title={t("bloodPressure", "Blood Pressure")}
+              onInputChange={(event: React.ChangeEvent<HTMLInputElement>) => {
+                event.target.name === "systolic"
+                  ? setPatientVitalAndBiometrics({
+                      ...patientVitalAndBiometrics,
+                      systolicBloodPressure: event.target.value
+                    })
+                  : setPatientVitalAndBiometrics({
+                      ...patientVitalAndBiometrics,
+                      diastolicBloodPressure: event.target.value
+                    });
+              }}
+              textFields={[
+                {
+                  name: t("systolic", "systolic"),
+                  separator: "/",
+                  type: "text",
+                  value: patientVitalAndBiometrics?.systolicBloodPressure || ""
+                },
+                {
+                  name: t("diastolic", "diastolic"),
+                  type: "text",
+                  value: patientVitalAndBiometrics?.diastolicBloodPressure || ""
+                }
+              ]}
+              unitSymbol={bloodPressureUnit}
+              isValidRange={
```

Comment:
```suggestion
              inputIsValid={
```

etc for all the other usages in this file